test: port-allocator 'returns different ports' is structurally racy and flakes CI

## Context

`client/test/_support/chain/port-allocator.test.ts` contains a test that is structurally racy and flakes intermittently on CI. It just blocked PR #705 (a documentation-only diff that doesn't touch any test or chain code):

```
FAIL test/_support/chain/port-allocator.test.ts > allocateAnvilPort > returns different ports on repeated calls
AssertionError: expected 34897 not to be 34897 // Object.is equality
```

## Why it's racy

`allocateAnvilPort()` (`client/test/_support/chain/port-allocator.ts`) binds to `:0`, lets the kernel pick a free port, reads the port number, then *closes the socket*. The function's own JSDoc admits:

> Caller must spawn its process promptly — the port is re-usable immediately but a racing allocator could pick it.

The test then does:
```ts
const a = await allocateAnvilPort();
const b = await allocateAnvilPort();
expect(a).not.toBe(b);
```

Between `a` and `b`, the first socket is closed and the kernel is free to return the same port on the next `bind(0)`. On Linux this is increasingly likely under `TIME_WAIT` reuse, low ephemeral-port-range pressure, or just regular kernel allocation policy. The test premise — "kernel always returns a different port on the next bind" — is wrong.

## Impact

Wastes contributor time. Triggers a CI rerun for any PR unlucky enough to hit it. False-failure-rate is low but not zero (~1 in N runs based on the fact that #705's first run failed and the next-branch CI has been green recently).

## Acceptance criteria

Pick one:

1. **Drop the `'returns different ports on repeated calls'` test.** The premise can't be made true without changing the allocator's contract (e.g. tracking issued ports in process-local state). And the existing first test — `'returns a listenable port'` — covers the function's actual contract.

2. **Make the test loop until it gets a different port.** Acceptable but adds latency to the CI run for no real value.

3. **Make `allocateAnvilPort` track issued ports across calls** and re-try if it would return a duplicate within the process. Adds real value (callers no longer need to "spawn their process promptly") but is a wider behavioural change.

My instinct is (1) — the second test isn't pulling its weight and the function's stated contract is honest about port reuse.

## Run-mode

test

--- a/client/test/_support/chain/port-allocator.test.ts
+++ b/client/test/_support/chain/port-allocator.test.ts
@@ -12,10 +12,4 @@ describe('allocateAnvilPort', () => {
       s.listen(port, '127.0.0.1', () => s.close(() => resolve()));
     });
   });
-
-  it('returns different ports on repeated calls', async () => {
-    const a = await allocateAnvilPort();
-    const b = await allocateAnvilPort();
-    expect(a).not.toBe(b);
-  });
 });

